stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ViewDescriptor.java

```diff
@@ -72,6 +124,125 @@ public View getViewForHighlighting(Object element) {
+        if (method.getName().equals("getId")) {
+          cssValue = getIdAttribute(element);
+
+          if (cssValue == null) {
+            cssValue = "(none)";
+          }
+        } else {
+          if (method.getReturnType() == int.class) {
+            int intValue = (int) returnedValue;
+
+            if (canIntBeMappedToString(annotation)) {
+              cssValue += " (" + mapIntToStringUsingAnnotations(intValue, annotation) + ")";
+              isValueDefault = false;
+            } else {
+              isValueDefault |= isNumberDefault(intValue, annotation);
+            }
+          } else if (method.getReturnType() == float.class) {
+            isValueDefault |= isNumberDefault((Number)returnedValue, annotation);
+          }
+        }
+
+        styles.store(new StyleInfo(cssName, cssValue, isValueDefault));
+      } catch (IllegalAccessException e) {
+        caught = true;
+      } catch (InvocationTargetException e) {
+        caught = true;
+      } finally {
+        if (caught) {
+          LogUtil.w("failed to get the style of element id=" +
+              getIdAttribute(element));
+        }
+      }
+    }
+  }
+
+  private static boolean canIntBeMappedToString(
+      ViewDebug.ExportedProperty annotation) {
+    return annotation != null
+           && annotation.mapping().length > 0;
+  }
+
+  private static String mapIntToStringUsingAnnotations(
+      int value,
+      ViewDebug.ExportedProperty annotation) {
+    if (!canIntBeMappedToString(annotation)) {
+      throw new IllegalStateException("Cannot map using this annotation");
+    }
+
+    for (ViewDebug.IntToString map : annotation.mapping()) {
+      if (map.from() == value) {
+        return map.to();
+      }
+    }
+
+    // no mapping was found even though one was expected ):
+    return " <no mapping> ";
+  }
+
+  private static boolean isNumberDefault(Number value, ViewDebug.ExportedProperty annotation) {
+    if (annotation != null) {
+      if (annotation.mapping().length > 0) {
+        return false;
+      }
+    }
+
+    return value.floatValue() == 0.0;
+  }
+
+  private static boolean isMethodGetter(Method method) {
+    return method.getName().toLowerCase().startsWith("get")
+           && method.getParameterTypes().length == 0;
+  }
+
+  @Nullable
+  private static ViewDebug.ExportedProperty getExportedAnnotation(Method method) {
+    Annotation[] annotations = method.getAnnotations();
+
+    for (Annotation annotation : annotations) {
+      if (annotation instanceof ViewDebug.ExportedProperty) {
+        return (ViewDebug.ExportedProperty) annotation;
+      }
+    }
+
+    return null;
+  }
+
+  private static String convertGetterNameToCssName(String getterName) {
+    // Split string by uppercase characters. Thankfully since
+    // this is the android source we don't have to worry about
+    // internationalization funk.
+    String[] words = getterName.split("(?=\\p{Upper})");
```

Comment:
because `String.split` uses a regex, this means we're compiling a regex every single time this method is called! I'm not sure what a better approach is just yet, but we will probably need to optimize this. Even if we just create our own method that doesn't use regex and uses brute force loops and stuff. (and then the output could also be funneled into an `Accumulator` and thus allow us to save lots of temporary allocations :) )